src/patterns/utils.rs

```diff
@@ -7,18 +7,15 @@ use libxml::tree::*;
  */
 
 /// checks whether a node is a comment node
-pub fn is_comment_node(node: &Node) -> bool {
-  node.get_type().unwrap() == NodeType::CommentNode
-}
+pub fn is_comment_node(node: &Node) -> bool { node.get_type().unwrap() == NodeType::CommentNode }
 
-/// gets the text content of a node. Requires that only child of the node is a text node
+/// gets the text content of a node. Requires that only child of the node is a
+/// text node
 pub fn get_simple_node_content(node: &Node, trim: bool) -> Result<String, String> {
   let child = node.get_first_child();
   if child.is_none() {
     Ok(String::new())
-  } else if !child.as_ref().unwrap().is_text_node()
-    || child.as_ref().unwrap().get_next_sibling().is_some()
-  {
+  } else if !child.as_ref().unwrap().is_text_node() || child.as_ref().unwrap().get_next_sibling().is_some() {
```

Comment:
Hm, this is a good example w.r.t readability. Maybe 200 is too large for a max width?